src/Fantomas/CodePrinter.fs

```diff
@@ -3000,6 +2954,192 @@ and genIndexers astContext node =
+
+        let addSpace = indentIfNeeded sepSpace
+
+        let genEx e =
+            if isCompExpr e then
+                sepSpace
+                +> sepOpenSFixed
+                +> sepSpace
+                +> indent
+                +> appNlnFun e (genExpr astContext e)
+                +> unindent
+            else
+                genExpr astContext e
+
+        atCurrentColumn
+            (genExpr astContext e
+             +> addFirstSpace
+             +> col addSpace es genEx)
+
+    let hasASingleInfixExpression =
+        match es with
+        | [ Paren (_, InfixApp _, _) ] -> true
+        | _ -> false
+
+    let isParenLambda =
+        (function
+        | Paren (_, Lambda _, _) -> true
+        | _ -> false)
+
+    let shouldHaveAlternativeLambdaStyle =
+        let hasLambdas = List.exists isParenLambda es
+
+        ctx.Config.AlternativeMultilineLambda
+        && (hasLambdas || hasASingleInfixExpression)
+
+
+    let longExpression =
+        if shouldHaveAlternativeLambdaStyle then
+            let hasMultipleArguments = (List.length es) > 1
+
+            let singleInfix =
+                col sepNone es (fun e ->
+                    let sepOpenTFor r =
+                        tokN (Option.defaultValue e.Range r) LPAREN sepOpenT
+
+                    let sepCloseTFor r =
+                        tokN (Option.defaultValue e.Range r) RPAREN sepCloseT
+
+                    match e with
+                    | Paren (lpr, (InfixApp _ as ia), rpr) ->
+                        let infixExpr = genExpr astContext ia
+
+                        let short =
+                            sepOpenTFor (Some lpr)
+                            +> infixExpr
+                            +> sepCloseTFor rpr
+
+                        let long =
+                            sepOpenTFor (Some lpr)
+                            +> indent
+                            +> sepNln
+                            +> infixExpr
+                            +> unindent
+                            +> sepNln
+                            +> sepCloseTFor rpr
+
+                        expressionFitsOnRestOfLine short long
+                    | _ -> genExpr astContext e)
+
+            let multipleArguments =
+                col sepNln es (fun e ->
+                    let genLambda pats (bodyExpr: SynExpr) lpr rpr arrowRange =
+                        let sepOpenTFor r =
+                            tokN (Option.defaultValue e.Range r) LPAREN sepOpenT
+
+                        let sepCloseTFor r =
+                            tokN (Option.defaultValue e.Range r) RPAREN sepCloseT
+
+                        let astCtx =
+                            { astContext with
+                                  IsInsideDotGet = false }
+
+                        let genExprAfterArrow (e: SynExpr) ctx =
+                            if Option.isSome (findTriviaTokenFromName RARROW e.Range ctx)
```

Comment:
I generally prefer the following just because it's a bit less prone to causing boolean blindness. But if this is the prevailing style then keep it.

```fsharp
match foo with
| Some _ -> blah
| None -> blah
```